### ISSUE TITLE
Wargus crashing after custom game.

### ISSUE THREAD

**Issue by JKCougar** [Wargus/wargus#434]
**Describe the bug**
After doing a custom game and winning or losing the game crashes with an error.

**To Reproduce**
1. Start a Standard Game
2. Go to Skirmish Modern
3. Setup a game between myself and AI
4. Finish the game
5. Click on Victory
6. See error

**Expected behavior**
It should pull up the final game stats 

**Screenshots and Logs**
![image](https://user-images.githubusercontent.com/121499276/209686417-57b16f0c-1e20-4174-b7b8-19d5a971c07c.png)

[stderr.txt](https://github.com/Wargus/wargus/files/10309283/stderr.txt)
[stdout.txt](https://github.com/Wargus/wargus/files/10309284/stdout.txt)
[Wargus-extraction.log](https://github.com/Wargus/wargus/files/10309285/Wargus-extraction.log)

Add screenshots to help explain your problem. If you encountered the issue inside a game, please add the log from the game (find it in "My Documents/Stratagus/wc2/logs" on Windows or "$HOME/.stratagus/wc2/logs" on Linux). Please also attach your stderr and stdout. If you're on Windows, these will be in "My Documents/Stratagus/std*.txt"

**Desktop (please complete the following information):**
 - OS: Windows 10 Professional
 - Version 21H2 19044.2364

**Comment by timfel** [comment]
No idea, sorry can you reproduce this reliably? There's nothing in the logs pointing to anything wrong.

**Comment by JKCougar** [comment]
> No idea, sorry can you reproduce this reliably? There's nothing in the logs pointing to anything wrong.

Yes it happens on every single custom game I do.  Would the version of my CDs make a difference? They are pretty old and probably not the latest version of WC2 from Blizzard.

**Comment by dr** [comment]
Same issue with 3.3.2 on Windows 11 22H2.  It only presents on custom games and initially begins as the game slowing down (music stuttering, gameplay lagging) before crashing.  If there's any logs I can provide, I'd be glad to.

**Comment by Hunteil** [comment]
> No idea, sorry can you reproduce this reliably? There's nothing in the logs pointing to anything wrong.

This issue has been happening to us over in discord chat. You got me and another separate active user. We could possibly get you logs to look through. I have it installed on a win 10 and a win 7 laptop (trying to play locally). Same version of game GoG and Wargus. Trying to play on local network. We can join each other but this same error crashes me on join win 10. I tried the campaign single player on windows 7 and it shows up and kills the game. It runs on win 10 in single player but when you leave the game on both OS the error shows up too. So it crashes going out the door too lol. Not sure what's going on here.

**Comment by SimoneStarace** [comment]
@Hunteil 

> I tried the campaign single player on windows 7 and it shows up and kills the game.

Does it happen on specific maps of the campaign or in any one? 
I tried with the first two human missions and the game didn't crash.

**Comment by sidav** [comment]
Update: strangely enough, I still get the error message in Wargus Portable run under wine, but the game does not crash. I finish the custom game as usual, exit the game via the menu... and see this error message after the game has exited. It does not occur if the game was exited without playing the custom game.

**Comment by SimoneStarace** [comment]
Hello @sidav.
Wargus version 3.3.2 has a weird bug that after finishing the map and pressing the victory button it crashes the game with an error message like the one in this issue.
Using a terminal to run Wargus it actually shows something more, like an error related to a double free memory corruption or something.
Unfortunately, I didn't test the nightly builds of Stratagus to see if this bug is still present.

**Comment by timfel** [comment]
Fixed
